vcd/resource_vcd_vapp_vm_hot_updates_test.go

```diff
@@ -0,0 +1,225 @@
+
+					resource.TestCheckResourceAttr("vcd_vapp_vm."+hotVmName1, "cpu_hot_add_enabled", "true"),
+					resource.TestCheckResourceAttr("vcd_vapp_vm."+hotVmName1, "memory_hot_add_enabled", "true"),
+
+					resource.TestCheckResourceAttr("vcd_vapp_vm."+hotVmName1, "memory", "2048"),
+					resource.TestCheckResourceAttr("vcd_vapp_vm."+hotVmName1, "cpus", "1"),
+				),
+			},
+			// Step 1 - update
+			resource.TestStep{
+				Config: configTextVMUpdateStep1,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckVcdVAppVmExists(hotVappName, hotVmName1, "vcd_vapp_vm."+hotVmName1, &vapp, &vm),
+					resource.TestCheckResourceAttr("vcd_vapp_vm."+hotVmName1, "name", hotVmName1),
+
+					resource.TestCheckResourceAttr("vcd_vapp_vm."+hotVmName1, "cpu_hot_add_enabled", "true"),
+					resource.TestCheckResourceAttr("vcd_vapp_vm."+hotVmName1, "memory_hot_add_enabled", "true"),
+
+					resource.TestCheckResourceAttr("vcd_vapp_vm."+hotVmName1, "memory", "3072"),
+					resource.TestCheckResourceAttr("vcd_vapp_vm."+hotVmName1, "cpus", "3"),
+					testAccCheckVcdVmNotRestarted("vcd_vapp_vm."+hotVmName1, hotVappName, hotVmName1),
+				),
+			},
+			resource.TestStep{
+				Config:      configTextVMUpdateStep2,
+				ExpectError: regexp.MustCompile(`update stopped: VM needs to reboot to change properties.*`),
+			},
+		},
+	})
+}
+
+func testAccCheckVcdVmNotRestarted(n string, vappName, vmName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no vApp ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*VCDClient)
+		org, vdc, err := conn.GetOrgAndVdc(testConfig.VCD.Org, testConfig.VCD.Vdc)
+		if err != nil {
+			return fmt.Errorf(errorRetrievingVdcFromOrg, testConfig.VCD.Vdc, testConfig.VCD.Org, err)
+		}
+
+		vapp, err := vdc.GetVAppByName(vappName, false)
+		if err != nil {
+			return err
+		}
+
+		vm, err := vapp.GetVMByName(vmName, false)
+		if err != nil {
+			return err
+		}
+
+		tasks, err := org.GetTaskList()
+		if err != nil {
+			return err
+		}
+
+		for _, task := range tasks.Task {
+			if strings.Contains(task.Operation, "Stopped") && task.Owner.ID == vm.VM.ID {
+				return fmt.Errorf("found task which stopped VM")
+			}
+		}
+
+		return nil
+	}
+}
+
+const testAccCheckVcdVAppHotUpdateVm = `
+resource "vcd_vapp" "{{.VAppName}}" {
+  org = "{{.Org}}"
+  vdc = "{{.Vdc}}"
+
+  name       = "{{.VAppName}}"
+}
+
+resource "vcd_vapp_vm" "{{.VMName}}" {
+  org = "{{.Org}}"
+  vdc = "{{.Vdc}}"
+
+  power_on = true
+
+  vapp_name     = vcd_vapp.{{.VAppName}}.name
+  name          = "{{.VMName}}"
+  computer_name = "compNameUp"
+
+  catalog_name  = "{{.Catalog}}"
+  template_name = "{{.CatalogItem}}"
+
+  memory        = 2048
+  cpus          = 1
+
+  cpu_hot_add_enabled    = true
+  memory_hot_add_enabled = true
+
```

Comment:
Nit: extra line.

Reply:
no, that's how I wanted. Separated by line.